parserator_web/views.py

```diff
@@ -14,11 +15,41 @@ class AddressParse(APIView):
     renderer_classes = [JSONRenderer]
 
     def get(self, request):
-        # TODO: Flesh out this method to parse an address string using the
-        # parse() method and return the parsed components to the frontend.
-        return Response({})
+        """Returns the string the user sent, the parsed address
+        components and the address type if the string is a valid address"""
+
+        # Get the unparsed address from JSON object query
+        input_string = request.query_params['address']
+
+        try:
+
+            # Get address components using .parse() method
+            address_components, address_type = self.parse(input_string)
+
+            # Return a dictionary object
+            return Response({
+                'input_string': input_string,
+                'address_components': address_components,
+                'address_type': address_type
+            })
+
+        except Exception as e:
+
+            # Return the expected dictionary object, but include error info
+            # and set HTTP status to 400
+            return Response({
+                'input_string': input_string,
+                'address_components': '',
+                'address_type': 'Invalid',
+                'error_message': str(e)},  # Currently not used.
+                status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Super smart to handle exceptions here! [Django REST Framework provides a `handle_exception` method](https://www.django-rest-framework.org/api-guide/views/#handle_exceptionself-exc) through which any errors raised by a dispatch method (like `get()`) are passed. Separating error handling into a different method enforces single responsibility for each function. 

By default, the `handle_exception` method only gracefully deals with REST Framework exceptions, plus a few Django exceptions. What do you think about extending the `handle_exception` method to handle expected errors, such as parse failures? (Basically, that would entail moving this logic into the extended method, very little new code required!)

Reply:
That makes sense! I think you're suggesting something like [this](https://github.com/Wizardess/code-challenge/pull/1/commits/bf0605dbc83d8d0a7fabffb8dae98c5163a8269a) - is that what you had in mind?